### PR TITLE
feat(FX-44390: Display recent price ranges

### PR DESCRIPTION
**Pull request by dimatretyak**
This PR resolves [FX-4439] 

#### iOS
https://user-images.githubusercontent.com/3513494/201910484-0f6cb22b-9c3b-4ca0-b129-d310ab278cfc.mp4

#### Android
https://user-images.githubusercontent.com/3513494/201918721-76452b5e-1d73-41dc-a510-45485b23ed6a.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [x] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Display recent price ranges - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[FX-4439]: https://artsyproduct.atlassian.net/browse/FX-4439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/ArtworkFilter/ArtworkFilterNavigator.tsx [hunk 1]
```diff
@@ -137,7 +140,25 @@ export const ArtworkFilterNavigator: React.FC<ArtworkFilterProps> = (props) => {
...
+
+    if (priceRange?.paramValue && priceRange.paramValue !== ParamDefaultValues.priceRange) {
+      /**
+       * We wait until the filter modal is closed, then save the recent price range
+       * Otherwise the recent price range will be displayed immediately after clicking on the "Show Results" button,
+       * Which can negatively affect the UI
+       */
+      setTimeout(() => {
+        GlobalStore.actions.recentPriceRanges.addNewPriceRange(priceRange.paramValue as string)
+      }, 500)
```
Comment (dimatretyak) [inline src/app/Components/ArtworkFilter/ArtworkFilterNavigator.tsx:156]: the problem that `setTimeout' solves


https://user-images.githubusercontent.com/3513494/201911329-146c735d-1d8d-4d1f-b38e-c69bb5076f25.mp4

File: src/app/store/AuthModel.ts [hunk 2]
```diff
@@ -365,7 +365,10 @@ export const getAuthModel = (): AuthModel => ({
       }
 
       if (user.id !== store.getState().previousSessionUserID) {
-        store.getStoreActions().search.clearRecentSearches()
+        const storeActions = store.getStoreActions()
+
+        storeActions.search.clearRecentSearches()
+        storeActions.recentPriceRanges.clearAllPriceRanges()
```
Comment (dimatretyak) [inline src/app/store/AuthModel.ts:371]: we clear the recent price ranges only if the user logs in to another account. If the user logs out and re-logs in to the same account, the recent price ranges will not be cleared. or is it better to always clear the recent price ranges when the user logs out of the account?

**Comment by dimatretyak** [comment]
**question**: should we hide these changes under feature flag? 🤔

**Comment by dblandin** [comment]
> question: should we hide these changes under feature flag? 🤔

I think a feature flag sounds like a good idea. When this merges and gets out to a beta build, we can have Ryan give this a design pass.

**Review by dblandin — APPROVED** [review]
LGTM 👍🏼

**Review by gkartalis — APPROVED** [review]
🥇 Also agree that a feature flag would make sense, other than that looks great!

**Pull request merged**